Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1401,19 +1401,22 @@ rewriteFile = rewriteFileEx normal
 -- the same as the existing content then leave the file as is so that we do not
 -- update the file's modification time.
 --
--- NB: the file is assumed to be ASCII-encoded.
+-- NB: Before Cabal-3.0 the file content was assumed to be
+--     ASCII-representable. Since Cabal-3.0 the file is assumed to be
+--     UTF-8 encoded.
 rewriteFileEx :: Verbosity -> FilePath -> String -> IO ()
 rewriteFileEx verbosity path newContent =
   flip catchIO mightNotExist $ do
-    existingContent <- annotateIO verbosity $ readFile path
-    _ <- evaluate (length existingContent)
-    unless (existingContent == newContent) $
+    existingContent <- annotateIO verbosity $ BS.Char8.readFile path
```

Comment:
Why use `BS.Char8` here and not `BS`?

Reply:
@23Skidoo no particular reason other than there was already a `BS.Char8` import but no `BS` import; but iirc I removed the last occurence using the Char8 API; so we could just drop the `.Char8` everywhere and things would work just fine.

Reply:
done